### PR TITLE
Wayland support using xdg-foreign

### PR DESCRIPTION
**Pull request by jadahl**
This needs gtk+ patches in https://bugzilla.gnome.org/show_bug.cgi?id=769788 and mutter patches in https://bugzilla.gnome.org/show_bug.cgi?id=769786 to function.

### CODE REVIEW

File: src/externalwindow-wayland.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+G_DEFINE_TYPE (ExternalWindowWayland, external_window_wayland,
+               EXTERNAL_TYPE_WINDOW)
+
+static GdkDisplay *
+get_wayland_display (void)
+{
+  if (wayland_display)
+    return wayland_display;
+
+  gdk_set_allowed_backends ("wayland");
```
Comment (matthiasclasen) [inline src/externalwindow-wayland.c]: To be pedantically correct, I guess we need to undo this after getting the display ?

File: src/externalwindow-wayland.c [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+static void
+external_window_wayland_set_parent_of (ExternalWindow *external_window,
+                                       GdkWindow      *child_window)
+{
+  ExternalWindowWayland *external_window_wayland =
+    EXTERNAL_WINDOW_WAYLAND (external_window);
+  char *handle_str = external_window_wayland->handle_str;
+
+  if (!gdk_wayland_window_set_transient_for_exported (child_window, handle_str))
```
Comment (matthiasclasen) [inline src/externalwindow-wayland.c]: This is not-yet-merged gdk-wayland api, right ? Probably needs a gtk version check, so we can build xdg-desktop-portal-gtk with 3.20, still.

Comment (jadahl) [inline src/externalwindow-wayland.c]: Sure. We can do that in configure.ac by adding >= 3.21.91 (or whatever version it'll be) after gtk+-wayland-3.0 in the PKG_CHECK_MODULES() thing.

File: src/externalwindow-wayland.c (hunk 1 above)
Comment (jadahl) [inline src/externalwindow-wayland.c]: Why not. Changed locally so far.

**Pull request merged**